test/s3_test.go

```diff
@@ -0,0 +1,234 @@
+package test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/require"
+)
+
+const output_bucket = "test-bucket"
+
+// initTestCases returns a list of BucketTestCase to be used for tests.
+func initTestCases() []BucketTestCase {
+	return []BucketTestCase{
+		{
+			testName: "TestBucket1",
+			vars: map[string]interface{}{
+				"test_bucket_name": "",
+				"read_only_paths":  []string{"path/to/ro-folder"},
+				"read_write_paths": []string{"path/to/rw-folder"},
+			},
+			objTestCases: []ObjectTestCase{
+				{
+					key:             "path/to/ro-folder/obj1",
+					encryption:      "AES256",
+					expectPassRead:  true,
+					expectPassWrite: false,
+				},
+				{
+					key:             "path/to/rw-folder/obj2",
+					encryption:      "AES256",
+					expectPassRead:  true,
+					expectPassWrite: true,
+				},
+				{
+					key:             "other/folder/obj3",
+					encryption:      "AES256",
+					expectPassRead:  false,
+					expectPassWrite: false,
+				},
+			},
+		},
+		{
+			testName: "TestBucket2",
```

Comment:
Let's make these test names more meaningful.

The big distinction here between TestBucket1 and TestBucket2 is the number of readable and writable paths defined in the policy, so let's capture that difference in the names.